web/client/utils/ConfigUtils.js

```diff
@@ -72,7 +72,7 @@ var ConfigUtils = {
         other.center = ConfigUtils.getCenter(other.center);
         return {
             map: other,
-            layers: layers.map(ConfigUtils.setBingKey),
+            layers: layers.map(ConfigUtils.setBingKey).map(ConfigUtils.setLayerId),
```

Comment:
So the id is mandatory and this function is an helper to assign the name if the id is not present.